### PR TITLE
#203 add sqoop task

### PR DESCRIPTION
**Pull request by zixi0825**
## What is the purpose of the pull request

*This pull request adds sqoop task.*

## Brief change log

  - *Add sqoop source type : mysql,hive,hdfs.*
  - *Add sqoop target type : mysql,hive,hdfs.*
  - *Add sqoop task ui.* 

## Verify this pull request

This change added tests and can be verified as follows:
  - *Added SqoopTaskTest to verify the change.*

### CODE REVIEW

**Comment by Eights-Li** [comment]
1 add license
2 add the test class to pom

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sqoop/SqoopTask.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import org.apache.dolphinscheduler.common.task.sqoop.SqoopParameters;
+import org.apache.dolphinscheduler.server.worker.task.TaskProps;
+import org.apache.dolphinscheduler.server.worker.task.shell.ShellTask;
+import org.apache.dolphinscheduler.server.worker.task.sqoop.generator.SqoopJobGenerator;
+import org.slf4j.Logger;
+
+/**
+ * sqoop task extends the shell task
+ */
+public class SqoopTask extends ShellTask {
```
Comment (Eights-Li) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sqoop/SqoopTask.java]: sqoop tasks will be converted to mapreduce tasks, managed by yarn， SqoopTask class need extends AbstractYarnTask not ShellTask

**Comment by Jave-Chen** [comment]
please make sure code have been checked by ali style plugin.

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/task/SqoopParameterEntityTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.apache.dolphinscheduler.common.task.sqoop.targets.TargetHiveParameter;
+import org.apache.dolphinscheduler.common.task.sqoop.targets.TargetMysqlParameter;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author simfo
```
Comment (Eights-Li) [inline dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/task/SqoopParameterEntityTest.java]: remove the author and time

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sqoop/generator/sources/MysqlSourceGenerator.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                            result.append(" --table ").append(sourceMysqlParameter.getSrcTable());
+                        }
+
+                        if(StringUtils.isNotEmpty(sourceMysqlParameter.getSrcColumns())){
+                            result.append(" --columns ").append(sourceMysqlParameter.getSrcColumns());
+                        }
+
+//                        StringBuilder srcWhere = new StringBuilder();
+//                        if(sourceMysqlParameter.getSrcConditionList().size()>0){
+//                            List<Property> items = sourceMysqlParameter.getSrcConditionList();
```
Comment (Eights-Li) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sqoop/generator/sources/MysqlSourceGenerator.java]: remove this comment out code

**Comment by Eights-Li** [comment]
@break60  please review front-end~

**Comment by break60** [comment]
+1

**Pull request merged**